### ISSUE TITLE
Make column information available to convertTo[PHP|Database]Value functions

### ISSUE THREAD

**Issue by michnovka** [doctrine/dbal#5341]
### Feature Request

I would like to get column options just like in `getSQLDeclaration()` also in `convertTo[PHP|Database]Value`

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no

#### Summary

I have defined my own `DecimalType` to use native PHP Decimals. The column declaration looks like

```php
#[ORM\Column(type: 'decimal', precision: 10, scale: 4)]
private Decimal $subtotal;
```

The `DecimalType` is below:

```php
use Decimal\Decimal;
use Doctrine\DBAL\Platforms\AbstractPlatform;
use Doctrine\DBAL\Types\DecimalType as DoctrineDecimalType;

class DecimalType extends DoctrineDecimalType
{
    public function getSQLDeclaration(array $column, AbstractPlatform $platform): string
    {
        $column['precision'] = empty($column['precision'])
            ? 10 : $column['precision'];
        $column['scale']     = empty($column['scale'])
            ? 0 : $column['scale'];

        return 'DECIMAL(' . $column['precision'] . ', ' . $column['scale'] . ')' .
            (!empty($column['unsigned']) ? ' UNSIGNED' : '');
    }

    public function convertToPHPValue($value, AbstractPlatform $platform)
    {
        if (is_null($value)) {
            return null;
        }

        return new Decimal($value, ???); // here precision would be useful
    }

    public function convertToDatabaseValue($value, AbstractPlatform $platform)
    {
        if ($value instanceof Decimal) {
            // HERE!! This is the line where I need to specify decimal precision based on column declaration of "scale"
            return $value->toFixed(????);
        }

        return parent::convertToDatabaseValue($value, $platform); // TODO: Change the autogenerated stub
    }
}
```

I need to format the value based on the scale/precision parameters from the Column definition.

Obviously a dirty hack would be to make several types to suite my needs like `decimal-10-6`, `decimal-10-4`,... But you can see why that is just a bad way to go.

I am sure there are many other examples where the column options would be useful to know when doing the casting. One that comes to my mind is unsigned operations.

[Here](https://stackoverflow.com/questions/71712912/how-to-get-column-properties-in-doctrine-custom-type-converttodatabasevalue-func?noredirect=1) is the ralated StackOverflow question I posted.

Thanks!

**Comment by morozov** [comment]
To me, this issue falls under the same "type parametrization" category as we discussed recently in https://github.com/doctrine/dbal/pull/4930. The `DecimalType` associated with a column should be parametrized with the values of precision and scale. Each instance of the `DecimalType` class should handle the `convertToPHPValue()` calls according to its configuration w/o that configuration being passed upon each call.

Also, linking to https://github.com/doctrine/dbal/issues/2841 for reference.

**Comment by michnovka** [comment]
@morozov I agree. These issues keep piling up in my project, parametrization would help with enumTypes as discussed in your referenced issue and would also be a solution to this: https://github.com/doctrine/orm/issues/9622
